### ISSUE TITLE
Use outlines instead ugly BoundingBox'es for selected MeshInstances (like in Unreal or Unity)

### ISSUE THREAD

**Issue by Chaosus** [godotengine/godot-proposals#2795]
### Describe the project you are working on
Godot

### Describe the problem or limitation you are having in your project
Not limitation just a visual improvement.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Currently, Godot using BoundingBoxes for selected meshes which I found kinda ugly:
![image](https://user-images.githubusercontent.com/3036176/119941032-bce52480-bf98-11eb-923b-47dab455c662.png)

My proposal is to change it to be like this:
![image](https://user-images.githubusercontent.com/3036176/119940958-a0e18300-bf98-11eb-8a8b-9a3028954543.png)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
Through shader I guess. Of course, this should be optional.

### If this enhancement will not be used often, can it be worked around with a few lines of script?
No.

### Is there a reason why this should be core and not an add-on in the asset library?
It's rendering stuff for the editor.

**Comment by Calinou** [comment]
reduz plans to work on this sometime in the future, but feel free to look into this if anyone is knowledgeable about it.

**Comment by UnknownS1024** [comment]
yeah the outline is better, but I didn't like AABB because it interferes with meshes for 3D.

**Comment by akien-mga** [comment]
We discussed this briefly in a proposal review meeting and agreed that this would be amazing, so feature wise this seems good.

We're not sure how this can be implemented in an easy way, would need input from @godotengine/rendering folks to see how this would be done and/or if it would be best waiting for having some support for stencil shaders.

**Comment by Calinou** [comment]
> IIRC there is already a PR/proposal (forgot which) for stencils

Indeed: https://github.com/godotengine/godot-proposals/issues/3373

**Comment by fire-forge** [comment]
What if an "Outline3D" node was added that adds an outline around a 3D object like what's shown in the Unreal Engine screenshot? Then the 3D editor could use it for outlining selected nodes, but it would also be really easy to add outlines in games.

**Comment by Calinou** [comment]
> What if an "Outline3D" node was added that adds an outline around a 3D object like what's shown in the Unreal Engine screenshot? Then the 3D editor could use it for outlining selected nodes, but it would also be really easy to add outlines in games.

This can already be achieved with the Material Overlay property. However, we can't override this property in the editor only, as its modified state could end up being saved in scene files.

**Comment by ryevdokimov** [comment]
I gave this a shot.

See: https://github.com/godotengine/godot/pull/109595
